src/Billing/Controllers/StripeController.cs

```diff
@@ -471,6 +480,71 @@ public async Task<IActionResult> PostWebhook([FromQuery] string key)
         return new OkResult();
     }
 
+    /// <summary>
+    /// Ensures that the customer associated with the parsed event's data is in the correct region for this server.
+    /// We use the customer instead of the subscription given that all subscriptions have customers, but not all
+    /// customers have subscriptions
+    /// </summary>
+    /// <param name="parsedEvent"></param>
+    /// <returns>true if the customer's region and the server's region match, otherwise false</returns>
+    /// <exception cref="Exception"></exception>
+    private async Task<bool> ValidateCloudRegionAsync(Event parsedEvent)
+    {
+        string customerRegion;
+
+        var serverRegion = _globalSettings.BaseServiceUri.CloudRegion;
+        var eventType = parsedEvent.Type;
+
+        switch (eventType)
+        {
+            case HandledStripeWebhook.SubscriptionDeleted:
+            case HandledStripeWebhook.SubscriptionUpdated:
+                {
+                    var subscription = await GetSubscriptionAsync(parsedEvent, true, new List<string>
+                {
+                    "customer"
+                });
+                    customerRegion = GetCustomerRegionFromMetadata(subscription.Customer.Metadata);
+                    break;
+                }
+            case HandledStripeWebhook.ChargeSucceeded:
+            case HandledStripeWebhook.ChargeRefunded:
+                {
+                    var charge = await GetChargeAsync(parsedEvent, true, new List<string> { "customer" });
+                    customerRegion = GetCustomerRegionFromMetadata(charge.Customer.Metadata);
+                    break;
+                }
+            case HandledStripeWebhook.UpcomingInvoice:
+            case HandledStripeWebhook.PaymentSucceeded:
+            case HandledStripeWebhook.PaymentFailed:
+            case HandledStripeWebhook.InvoiceCreated:
+                {
+                    var invoice = await GetInvoiceAsync(parsedEvent, true, new List<string> { "customer" });
+                    customerRegion = GetCustomerRegionFromMetadata(invoice.Customer.Metadata);
+                    break;
+                }
+            default:
+                {
+                    // For all Stripe events that we're not listening to, just return 200
+                    return false;
+                }
+        }
+
+        return customerRegion == serverRegion;
+    }
+
+    /// <summary>
+    /// Gets the region from the customer metadata. If no region is present, defaults to "US"
+    /// </summary>
+    /// <param name="customerMetadata"></param>
+    /// <returns></returns>
+    private static string GetCustomerRegionFromMetadata(Dictionary<string, string> customerMetadata)
+    {
+        return !customerMetadata.ContainsKey("region")
+            ? "US"
+            : customerMetadata["region"];
```

Comment:
Another nitpick: I prefer positive statements
```suggestion
        return customerMetadata.ContainsKey("region")
            ? customerMetadata["region"]
            : "US";
```

Reply:
Updated in the last push